### PR TITLE
refactor(frontend): minor changes around `FrontendInstance` constructor 

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

minor refactor of repaying 2 todos, and others
- get `CatalogManager` from the instance on constructing, rather than set it later
- move mock constructor to test utils 

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by killme2008 — APPROVED** [review]
LGTM

File: src/servers/src/postgres/server.rs [hunk 1]
```diff
@@ -79,6 +80,13 @@ impl PostgresServer {
                 match tcp_stream {
                     Err(error) => error!("Broken pipe: {}", error), // IoError doesn't impl ErrorExt.
                     Ok(io_stream) => {
+                        match io_stream.peer_addr() {
+                            Ok(addr) => info!("PostgreSQL connection coming from {}", addr),
```
Comment (waynexia) [inline src/servers/src/postgres/server.rs:84]: change this to `debug` or `trace` level? `info` might be annoying as we will `warn` on the error path.

File: src/servers/src/postgres/server.rs [hunk 2]
```diff
@@ -79,6 +80,11 @@ impl PostgresServer {
                 match tcp_stream {
                     Err(error) => error!("Broken pipe: {}", error), // IoError doesn't impl ErrorExt.
                     Ok(io_stream) => {
+                        match io_stream.peer_addr() {
```
Comment (v0y4g3r) [inline src/servers/src/postgres/server.rs:83]: Looks like it's just for debugging?

**Review by v0y4g3r — APPROVED** [review]
LGTM

File: src/servers/src/postgres/server.rs (hunk 2 above)
Comment (MichaelScofield) [reply src/servers/src/postgres/server.rs:83]: yes, very useful for dev

**Pull request merged**